stream.go

```diff
@@ -1002,18 +985,33 @@ func (cs *clientStream) finish(err error) {
 			}
 		}
 	}
+
 	cs.mu.Unlock()
 	// For binary logging. only log cancel in finish (could be caused by RPC ctx
```

Comment:
Now it logs cancel and server trailers :).

Reply:
Hmm, I think the logic here is a bit wrong, since the server could return a `Canceled` status with trailers which is distinct from the client initiating cancelation.  I'll have to work on this and add a test since I think it worked properly before.